### DISCUSSION TITLE
Janus Graph Extremely Slow on Docker

### DISCUSSION THREAD

**Discussion by raxelz** [JanusGraph/janusgraph#2328] [Q&A] [answered]
Hi everyone,
My name is Razi, i've been working on a graph service for a month now attempting to use JanusGraph

Current Service Setup: **My service -> Janus/Gremlin -> Cassandra (Disk)**

When running on metal, I get relatively good performance, but as soon as I switch to docker compose example, the performance goes down 10x and response time goes up.

Docker Resource Spec: 16 cores / 28 GB Mem/ 128 GB SSD
Janus Config: CQL / 16 Workers / 32 Janus Pool
Service Config: 300 Max connection pool size

I have been attempting to find the issue through different monitoring but I cant find it.
Did anyone run into this issue? Its been sometime while investigating, its getting gnarly

**Comment by mmadoo** [comment, marked as answer]
I run janusGraph on k8s, I have not seen difference with bare metal.
Which java options have you set (-Xmx, -Xms, ...) in both cases for JG and Cassandra ?

**Reply by raxelz** [reply]
We were running docker on MacOs, when running it on Linux it resolved the performance differences issue.
It was only virtualization on MacOs.

Thank you for the response!
